src/Mod/PartDesign/PartDesignTests/TestTopologicalNamingProblem.py

```diff
@@ -0,0 +1,136 @@
+# SPDX-License-Identifier: LGPL-2.1-or-later
+# ***************************************************************************
+# *                                                                         *
+# *   Copyright (c) 2024 bgbsww@gmail.com                                   *
+# *                                                                         *
+# *   This file is part of FreeCAD.                                         *
+# *                                                                         *
+# *   FreeCAD is free software: you can redistribute it and/or modify it    *
+# *   under the terms of the GNU Lesser General Public License as           *
+# *   published by the Free Software Foundation, either version 2.1 of the  *
+# *   License, or (at your option) any later version.                       *
+# *                                                                         *
+# *   FreeCAD is distributed in the hope that it will be useful, but        *
+# *   WITHOUT ANY WARRANTY; without even the implied warranty of            *
+# *   MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU      *
+# *   Lesser General Public License for more details.                       *
+# *                                                                         *
+# *   You should have received a copy of the GNU Lesser General Public      *
+# *   License along with FreeCAD. If not, see                               *
+# *   <https://www.gnu.org/licenses/>.                                      *
+# *                                                                         *
+# ***************************************************************************
+import unittest
+
+
+import FreeCAD as App
+import Part
+import Sketcher
+import TestSketcherApp
+
+class TestTopologicalNamingProblem(unittest.TestCase):
+    def setUp(self):
+        self.Doc = App.newDocument("PartDesignTestTNP")
+
+    def testPadsOnBaseObject(self):
```

Comment:
I like this test but I'm not sure about it's purpose:

The model defined here doesn't look prone to TNP issues (unless we do some operation that modifies referenced geometries, for example replacing `self.PadSketch` with a circle).

So is this test meant to be a check that the code imported for TNP doesn't introduce new issues?